erizo/src/erizo/rtp/SequenceNumberTranslator.cpp

```diff
@@ -0,0 +1,114 @@
+#include "rtp/SequenceNumberTranslator.h"
+
+#include <vector>
+
+namespace erizo {
+
+constexpr uint16_t kMaxSequenceNumberInBuffer = 511;  // = 65536 / 128 and more than 10s of audio @50fps
+constexpr uint16_t kMaxDistance = 200;
+
+DEFINE_LOGGER(SequenceNumberTranslator, "rtp.SequenceNumberTranslator");
+
+SequenceNumberTranslator::SequenceNumberTranslator()
+    : in_out_buffer_{kMaxSequenceNumberInBuffer},
+      out_in_buffer_{kMaxSequenceNumberInBuffer},
+      first_input_sequence_number_{0},
+      last_input_sequence_number_{0},
+      initial_output_sequence_number_{0},
+      initialized_{false}, reset_{false} {
+}
+
+bool SequenceNumberTranslator::lessThan(uint16_t first, uint16_t last) const {
+  uint16_t result = first - last;
+  return result > 0xF000;
+}
+
+void SequenceNumberTranslator::add(SequenceNumber sequence_number) {
+  in_out_buffer_[sequence_number.input % kMaxSequenceNumberInBuffer] = sequence_number;
+  out_in_buffer_[sequence_number.output % kMaxSequenceNumberInBuffer] = sequence_number;
+}
+
+uint16_t SequenceNumberTranslator::fill(const uint16_t &first,
+                                        const uint16_t &last) {
+  const SequenceNumber& first_sequence_number = get(first);
+  uint16_t output_sequence_number = first_sequence_number.output;
+
+  if (first_sequence_number.type != SequenceNumberType::Skip) {
+    output_sequence_number++;
+  }
+
+  for (uint16_t sequence_number = first_sequence_number.input + 1;
+       lessThan(sequence_number, last);
+       sequence_number++) {
+    add({sequence_number, output_sequence_number++, SequenceNumberType::Valid});
```

Comment:
we'll use `add(SequenceNumber{sequence_number, output_sequence_number++, SequenceNumberType::Valid});` as the best approach.